pkg/peaks/peaks.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peaks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+
+	"github.com/paypal/load-watcher/pkg/watcher"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	"sigs.k8s.io/scheduler-plugins/apis/config"
+	cfgv1 "sigs.k8s.io/scheduler-plugins/apis/config/v1"
+	"sigs.k8s.io/scheduler-plugins/pkg/trimaran"
+)
+
+const (
+	Name = "Peaks"
+)
+
+type Peaks struct {
+	handle    framework.Handle
+	collector *trimaran.Collector
+	args      *config.PeaksArgs
+}
+
+type PowerModel struct {
+	K0 float64 `json:"k0"`
+	K1 float64 `json:"k1"`
+	K2 float64 `json:"k2"`
+	// Power = K0 + K1 * e ^(K2 * x) : where x is utilisation
+}
+
+var _ framework.ScorePlugin = &Peaks{}
+var maxPower = 0.0
+var clusterPowerModel map[string]PowerModel
+
+func (pl *Peaks) Name() string {
+	return Name
+}
+
+func initNodePowerModels() error {
+	data, err := os.ReadFile("/power_model/node_power_model")
+	if err != nil {
+		klog.ErrorS(err, "Unable to read power model from configMap")
+		return err
+	}
+	if err = json.Unmarshal(data, &clusterPowerModel); err != nil {
+		klog.ErrorS(err, "Unable to unmarshal power model from configMap value")
+		return err
+	}
+	return nil
+}
+
+func New(_ context.Context, obj runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	klog.V(4).InfoS("Peaks plugin Input config %+v\n", obj)
```

Comment:
You don't need use append `\n`:

```suggestion
	klog.V(4).InfoS("Peaks plugin Input config %+v", obj)
```

Reply:
Removed newline.